### ISSUE TITLE
Imaginary part of real numbers is not plotted for ComplexSignal

### ISSUE THREAD

**Issue by sindrerb** [hyperspy/hyperspy#1893]
I have experienced that the imaginary part of a complex signal sometimes misses when plotted. It seems like it is consequently when the indices of the signal corresponds to a zero imaginary part. A reconstructed situation is added below where index 0 results in a missing imaginary part but index 1 does not.

I believe the bug is within the` plot() ` function in the `ComplexSignal1D` class, but I cannot see a clear cause.

```python
data = (np.ones((500,2)) + 1j*np.ones((500,2))*np.array([0,1])).T

signal = hs.signals.ComplexSignal1D(data)

# Imaginary part missing
#signal.axes_manager.indices = (0,)
#signal.plot()

# Imaginary part appear
signal.axes_manager.indices = (1,)
signal.plot()
```

**Comment by francisco-dlp** [comment]
The line is not missing, but simply hidden by the axes spines, just pan to see it.

**Comment by thomasaarholt** [comment]
@francisco-dlp that's not correct - @sindrerb's example is easier to see if you add a 2* multiplier to the imaginary part.

```python
data = (np.ones((500,2)) + 2*1j*np.ones((500,2))*np.array([0,1])).T

signal = hs.signals.ComplexSignal1D(data)

# Imaginary part missing
signal.axes_manager.indices = (0,)
signal.plot()

# Imaginary part appear
#signal.axes_manager.indices = (1,)
#signal.plot()
```
With the "missing" part commented in, you'll see that the data is in fact not visible in the plot.

Edit: In particular, if one runs the "missing" code, and then uses the arrow key to move to the second index, and then zoom out, you see that it is not present.

**Comment by thomasaarholt** [comment]
Even better example:
First run 
```python
length = 100
x = np.arange(0,length)
x = np.array([x,x])
real = np.sin(2*np.pi*x/length)
imag = (-1j*np.array([0,1])*real.T).T

data = real + imag
s = hs.signals.ComplexSignal1D(data)
```
Then run this followed by moving right arrow.
```python
s.axes_manager[0].index = 0
s.plot()
```
Finally run this to see the same index as was shown after right arrow in the previous code.
```python
s.axes_manager[0].index = 1
s.plot()
```

Whether one uses complexsignal1D or signal1D has no effect.

**Comment by ericpre** [comment]
A complex number with an imaginary part equal to zero is a real number, so hyperspy plots it as real! ;) However, when plotting a `ComplexSignal`, it is fair to expect to see the imaginary part even if the data is actually real and not complex. Fixed in #1896.

**Comment by thomasaarholt** [comment]
Great Scott, that was quick!

>

**Comment by francisco-dlp** [comment]
Fixed in #1896.
